packages/client/src/components/pages/muster-page/table-unit-muster-compliance.tsx

```diff
@@ -76,69 +75,34 @@ export const UnitMusterComplianceTable = (props: any) => {
 
   const reloadMusterComplianceData = useCallback(async () => {
     try {
-      console.log(units);
-      console.log(rosterUnitId);
-      console.log(rosterFromDateIso);
       // determine if we are displaying for all units or just a single unit
       const unitsToDisplay = rosterUnitId === -1 ? units : [units.find(unit => unit.id === rosterUnitId)];
-      const plotData = new Map();
-      const complianceDataByUnit = new Map();
+      const plotDataByUnit = new Map();
+      const complianceStatsByUnit = new Map();
```

Comment:
These resolve to `Map<any, any>`, losing type checking. I guess this is a slight downside of Map, where it will sneak that through. You can add typings to the Map, but maybe it's better to just use plain objects here.

Reply:
I'll add the typing, technically we get the check down in the code because we pass these as values to `setUnitComplianceStats` etc, which expect type conformity, I can make these temp variables explicit though.

Reply:
I think since these default to `any` they'll actually pass those type checks as well. When a type is `any` it just passes all type checks. So you could potentially pass erroneous data through that wouldn't actually conform to the state values.

Reply:
Wow I thought .ts was a bit better at this, you are right! I don't even get a runtime error which is super disturbing even if the ambiguous map contains invalid data key or value data types. The lack of runtime error is actually a head scratcher for me...I'll have to brush up on my .ts a bit more it looks like. Either way the maps are typed now.

Reply:
We have the strict settings enabled, so implicit `any` types aren't allowed. But in this case they snuck through because they're just template types that default to `any`. The default map constructor looks like this: `Map<any, any>`.

That's one of the reasons I'm not sure how much I like Map. I also just sort of prefer staying in pure JSON as much of the time as possible, since that's the format that's being serialized and shuttled around everywhere.